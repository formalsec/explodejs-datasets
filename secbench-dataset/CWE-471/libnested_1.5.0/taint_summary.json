[
    {
        "type": "VFunPropOfExportedObj",
        "filename": "/home/filipe/projects/explode-js/bench/datasets/secbench-dataset/CWE-471/libnested_1.5.0/src/index.js",
        "vuln_type": "prototype-pollution",
        "sink": "obj[path[i]] = value",
        "sink_lineno": 27,
        "source": "module.exports.set",
        "tainted_params": [
            "obj",
            "path",
            "value"
        ],
        "params_types": {
            "obj": "object",
            "path": {
                "_union": [
                    "object",
                    "array",
                    "string"
                ]
            },
            "value": "any"
        },
        "call_paths": [
            {
                "type": "Method",
                "prop": "set",
                "fn_name": "1597.set-o210",
                "fn_id": "1072",
                "source_fn_id": "1072"
            }
        ]
    }
]
