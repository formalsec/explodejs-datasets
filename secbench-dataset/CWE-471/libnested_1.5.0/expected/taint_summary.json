[
  {
    "type": "VFunPropOfExportedObj",
    "filename" : "index.js",
    "vuln_type": "prototype-pollution",
    "sink": "obj[path[i]] = value",
    "sink_lineno": 27,
    "source": "module.exports.set",
    "tainted_params": [ "obj", "path", "value" ],
    "params_types": {
      "obj": "object",
      "path" : [ "string", "string" ],
      "value" : "string"
    }
  }
]
